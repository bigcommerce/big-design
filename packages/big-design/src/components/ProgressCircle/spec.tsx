import 'jest-styled-components';

import { render } from '@testing-library/react';
import React from 'react';

import { ProgressCircle } from './ProgressCircle';

test('render determinant progress circle', () => {
  const { container } = render(<ProgressCircle percent={50} />);

  expect(container).toMatchSnapshot();
});

test('render completed determinant progress circle', () => {
  const { container } = render(<ProgressCircle percent={100} />);

  expect(container).toMatchSnapshot();
});

test('render error determinant progress circle', () => {
  const { container } = render(<ProgressCircle error={true} percent={0} />);

  expect(container).toMatchSnapshot();
});

test('render indeterminant progress bar', () => {
  const { container } = render(<ProgressCircle />);

  expect(container).toMatchSnapshot();
});

test('render large progress circle', () => {
  const { container } = render(<ProgressCircle size="large" />);

  expect(container).toMatchSnapshot();
});

test('render medium progress circle', () => {
  const { container } = render(<ProgressCircle size="medium" />);

  expect(container).toMatchSnapshot();
});

test('render small progress circle', () => {
  const { container } = render(<ProgressCircle size="small" />);

  expect(container).toMatchSnapshot();
});

test('render xSmall progress circle', () => {
  const { container } = render(<ProgressCircle size="xSmall" />);

  expect(container).toMatchSnapshot();
});

test('render xSmall progress circle', () => {
  const { container } = render(<ProgressCircle size="xxSmall" />);

  expect(container).toMatchSnapshot();
});
