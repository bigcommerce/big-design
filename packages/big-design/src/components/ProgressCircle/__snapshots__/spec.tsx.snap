// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`render completed determinant progress circle 1`] = `
.c0 {
  vertical-align: middle;
  color: #2AAB3F;
  height: 3rem;
  width: 3rem;
}

<div>
  <svg
    aria-hidden="true"
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="100"
    class="c0"
    color="success"
    fill="currentColor"
    height="24"
    role="progressbar"
    size="48"
    stroke="currentColor"
    stroke-width="0"
    viewBox="0 0 24 24"
    width="24"
  >
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
    <path
      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2M9.29 16.29 5.7 12.7a.996.996 0 1 1 1.41-1.41L10 14.17l6.88-6.88a.996.996 0 1 1 1.41 1.41l-7.59 7.59a.996.996 0 0 1-1.41 0"
    />
  </svg>
</div>
`;

exports[`render determinant progress circle 1`] = `
.c0 {
  height: 3rem;
  width: 3rem;
}

.c1 {
  fill: transparent;
  stroke-width: 0.25rem;
  stroke: #ECEEF5;
}

.c2 {
  stroke-dasharray: 138.23007675795088 138.23007675795088;
  stroke: #3C64F4;
  -webkit-transform-origin: 50% 50%;
  -ms-transform-origin: 50% 50%;
  transform-origin: 50% 50%;
  stroke-dashoffset: 69.11503837897544;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
  -webkit-transition: stroke-dashoffset 0.35s;
  transition: stroke-dashoffset 0.35s;
}

.c3 {
  font-size: 0.875rem;
  font-weight: 400;
}

<div>
  <svg
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="50"
    class="c0"
    role="progressbar"
  >
    <circle
      class="c1"
      cx="1.5em"
      cy="1.5em"
      r="1.375em"
    />
    <circle
      class="c1 c2"
      cx="1.5em"
      cy="1.5em"
      r="1.375em"
    />
    <text
      class="c3"
      dominant-baseline="central"
      text-anchor="middle"
      x="50%"
      y="50%"
    >
      50
      %
    </text>
  </svg>
</div>
`;

exports[`render error determinant progress circle 1`] = `
.c0 {
  vertical-align: middle;
  color: #DB3643;
  height: 3rem;
  width: 3rem;
}

<div>
  <svg
    aria-hidden="true"
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="0"
    aria-valuetext="Error"
    class="c0"
    color="danger"
    fill="currentColor"
    height="24"
    role="progressbar"
    size="48"
    stroke="currentColor"
    stroke-width="0"
    viewBox="0 0 24 24"
    width="24"
  >
    <path
      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 11c-.55 0-1-.45-1-1V8c0-.55.45-1 1-1s1 .45 1 1v4c0 .55-.45 1-1 1m1 4h-2v-2h2z"
    />
  </svg>
</div>
`;

exports[`render indeterminant progress bar 1`] = `
.c0 {
  height: 3rem;
  width: 3rem;
}

.c1 {
  fill: transparent;
  stroke-width: 0.25rem;
  stroke: #ECEEF5;
}

.c2 {
  stroke-dasharray: 138.23007675795088 138.23007675795088;
  stroke: #3C64F4;
  -webkit-transform-origin: 50% 50%;
  -ms-transform-origin: 50% 50%;
  transform-origin: 50% 50%;
  -webkit-animation: dfVSSu 1s ease-out infinite;
  animation: dfVSSu 1s ease-out infinite;
  stroke-dashoffset: 138.23007675795088;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
  -webkit-transition: stroke-dashoffset 0.35s;
  transition: stroke-dashoffset 0.35s;
}

<div>
  <svg
    class="c0"
    role="progressbar"
  >
    <circle
      class="c1"
      cx="1.5em"
      cy="1.5em"
      r="1.375em"
    />
    <circle
      class="c1 c2"
      cx="1.5em"
      cy="1.5em"
      r="1.375em"
    />
  </svg>
</div>
`;

exports[`render large progress circle 1`] = `
.c0 {
  height: 5rem;
  width: 5rem;
}

.c1 {
  fill: transparent;
  stroke-width: 0.5rem;
  stroke: #ECEEF5;
}

.c2 {
  stroke-dasharray: 226.1946710584651 226.1946710584651;
  stroke: #3C64F4;
  -webkit-transform-origin: 50% 50%;
  -ms-transform-origin: 50% 50%;
  transform-origin: 50% 50%;
  -webkit-animation: krEchf 1s ease-out infinite;
  animation: krEchf 1s ease-out infinite;
  stroke-dashoffset: 226.1946710584651;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
  -webkit-transition: stroke-dashoffset 0.35s;
  transition: stroke-dashoffset 0.35s;
}

<div>
  <svg
    class="c0"
    role="progressbar"
  >
    <circle
      class="c1"
      cx="2.5em"
      cy="2.5em"
      r="2.25em"
    />
    <circle
      class="c1 c2"
      cx="2.5em"
      cy="2.5em"
      r="2.25em"
    />
  </svg>
</div>
`;

exports[`render medium progress circle 1`] = `
.c0 {
  height: 3rem;
  width: 3rem;
}

.c1 {
  fill: transparent;
  stroke-width: 0.25rem;
  stroke: #ECEEF5;
}

.c2 {
  stroke-dasharray: 138.23007675795088 138.23007675795088;
  stroke: #3C64F4;
  -webkit-transform-origin: 50% 50%;
  -ms-transform-origin: 50% 50%;
  transform-origin: 50% 50%;
  -webkit-animation: dfVSSu 1s ease-out infinite;
  animation: dfVSSu 1s ease-out infinite;
  stroke-dashoffset: 138.23007675795088;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
  -webkit-transition: stroke-dashoffset 0.35s;
  transition: stroke-dashoffset 0.35s;
}

<div>
  <svg
    class="c0"
    role="progressbar"
  >
    <circle
      class="c1"
      cx="1.5em"
      cy="1.5em"
      r="1.375em"
    />
    <circle
      class="c1 c2"
      cx="1.5em"
      cy="1.5em"
      r="1.375em"
    />
  </svg>
</div>
`;

exports[`render small progress circle 1`] = `
.c0 {
  height: 2rem;
  width: 2rem;
}

.c1 {
  fill: transparent;
  stroke-width: 0.25rem;
  stroke: #ECEEF5;
}

.c2 {
  stroke-dasharray: 87.96459430051421 87.96459430051421;
  stroke: #3C64F4;
  -webkit-transform-origin: 50% 50%;
  -ms-transform-origin: 50% 50%;
  transform-origin: 50% 50%;
  -webkit-animation: kpuuIv 1s ease-out infinite;
  animation: kpuuIv 1s ease-out infinite;
  stroke-dashoffset: 87.96459430051421;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
  -webkit-transition: stroke-dashoffset 0.35s;
  transition: stroke-dashoffset 0.35s;
}

<div>
  <svg
    class="c0"
    role="progressbar"
  >
    <circle
      class="c1"
      cx="1em"
      cy="1em"
      r="0.875em"
    />
    <circle
      class="c1 c2"
      cx="1em"
      cy="1em"
      r="0.875em"
    />
  </svg>
</div>
`;

exports[`render xSmall progress circle 1`] = `
.c0 {
  height: 1.5rem;
  width: 1.5rem;
}

.c1 {
  fill: transparent;
  stroke-width: 0.1875rem;
  stroke: #ECEEF5;
}

.c2 {
  stroke-dasharray: 65.97344572538566 65.97344572538566;
  stroke: #3C64F4;
  -webkit-transform-origin: 50% 50%;
  -ms-transform-origin: 50% 50%;
  transform-origin: 50% 50%;
  -webkit-animation: hiqbsE 1s ease-out infinite;
  animation: hiqbsE 1s ease-out infinite;
  stroke-dashoffset: 65.97344572538566;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
  -webkit-transition: stroke-dashoffset 0.35s;
  transition: stroke-dashoffset 0.35s;
}

<div>
  <svg
    class="c0"
    role="progressbar"
  >
    <circle
      class="c1"
      cx="0.75em"
      cy="0.75em"
      r="0.65625em"
    />
    <circle
      class="c1 c2"
      cx="0.75em"
      cy="0.75em"
      r="0.65625em"
    />
  </svg>
</div>
`;

exports[`render xSmall progress circle 2`] = `
.c0 {
  height: 1rem;
  width: 1rem;
}

.c1 {
  fill: transparent;
  stroke-width: 0.125rem;
  stroke: #ECEEF5;
}

.c2 {
  stroke-dasharray: 43.982297150257104 43.982297150257104;
  stroke: #3C64F4;
  -webkit-transform-origin: 50% 50%;
  -ms-transform-origin: 50% 50%;
  transform-origin: 50% 50%;
  -webkit-animation: fxwtRW 1s ease-out infinite;
  animation: fxwtRW 1s ease-out infinite;
  stroke-dashoffset: 43.982297150257104;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
  -webkit-transition: stroke-dashoffset 0.35s;
  transition: stroke-dashoffset 0.35s;
}

<div>
  <svg
    class="c0"
    role="progressbar"
  >
    <circle
      class="c1"
      cx="0.5em"
      cy="0.5em"
      r="0.4375em"
    />
    <circle
      class="c1 c2"
      cx="0.5em"
      cy="0.5em"
      r="0.4375em"
    />
  </svg>
</div>
`;
